sklearn/calibration.py

```diff
@@ -431,19 +449,21 @@ def _fit_classifier_calibrator_pair(estimator, X, y, train, test, supports_sw,
     -------
     calibrated_classifier : _CalibratedClassifier instance
     """
+    X_train, y_train = _safe_indexing(X, train), _safe_indexing(y, train)
+    X_test, y_test = _safe_indexing(X, test), _safe_indexing(y, test)
     if sample_weight is not None and supports_sw:
-        estimator.fit(X[train], y[train],
-                      sample_weight=sample_weight[train])
+        estimator.fit(X_train, y_train,
+                      sample_weight=_safe_indexing(sample_weight, train))
```

Comment:
you could add `sw_train` and `sw_test` above to have all of the in the same place. Ignore if you think it doesn't make the code more readable.

Reply:
indeed, done.